be/src/exprs/agg/hll_sketch.h

```diff
@@ -78,26 +85,31 @@ class HllSketchAggregateFunction final
                 value = HashUtil::murmur_hash64A(&v[i], sizeof(v[i]), HashUtil::MURMUR_SEED);
 
                 if (value != 0) {
-                    this->data(state).update(value);
+                    this->data(state).hll_sketch->update(value);
                 }
             }
         }
     }
 
     void merge(FunctionContext* ctx, const Column* column, AggDataPtr __restrict state, size_t row_num) const override {
         DCHECK(column->is_binary());
-
         const BinaryColumn* hll_column = down_cast<const BinaryColumn*>(column);
         DataSketchesHll hll(hll_column->get(row_num).get_slice());
-        this->data(state).merge(hll);
+        if (UNLIKELY(this->data(state).hll_sketch == nullptr)) {
+            this->data(state).hll_sketch =
+                    std::make_unique<DataSketchesHll>(hll.get_lg_config_k(), hll.get_target_type());
+        }
+        this->data(state).hll_sketch->merge(hll);
     }
 
     void get_values(FunctionContext* ctx, ConstAggDataPtr __restrict state, Column* dst, size_t start,
                     size_t end) const override {
         DCHECK_GT(end, start);
         Int64Column* column = down_cast<Int64Column*>(dst);
-        int64_t result = this->data(state).estimate_cardinality();
-
+        int64_t result = 0L;
+        if (LIKELY(this->data(state).hll_sketch != nullptr)) {
```

Comment:
it seems that if a HLLSketchState is constructed, then HLLSketchState.hll_sketch is never nullptr.

do not worry about that low-cardinality HLLSketch occupies too many memory, accoding to its corresponding  Java's implementations, in this situations, array and hashset is adopted instead of too many buckets.